crates/rover-client/src/operations/graph/check_workflow/runner.rs

```diff
@@ -85,32 +86,48 @@ fn get_check_response_from_data(
         }
     }
 
-    if let Some(result) = operations_result {
+    if matches!(operations_status, Some(CheckWorkflowTaskStatus::FAILED))
+        || matches!(workflow_status, CheckWorkflowStatus::PASSED)
+    {
+        let result = operations_result.ok_or(RoverClientError::AdhocError {
+            msg: "Operations check task has no result.".to_string(),
```

Comment:
is there a better error we can return here/do we think it is unreachable/other changes here make this unlikely?

Reply:
If a check workflow has passed, then all its tasks must have passed, which includes the operations check task (so it must have a result). Similarly, if the operations check task has failed, then it also must have a result.

So if we encounter this message/case, it basically means there's a bug in our API, or the semantics of our task API have changed in a breaking way. (So at the very least, it should be unlikely.) What error would you recommend throwing in this case?